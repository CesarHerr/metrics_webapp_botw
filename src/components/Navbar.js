import { NavLink } from 'react-router-dom';

function Navbar() {
  return (
    <header>
      <nav>
        <h1>Api</h1>
        <NavLink to="/">Home</NavLink>
      </nav>
    </header>
  );
}
export default Navbar;
