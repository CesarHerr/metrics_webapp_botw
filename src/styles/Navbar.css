header {
  background-color: rgba(9, 138, 47);
  padding: 0.4rem;
  position: fixed;
  width: 100%;
  z-index: 1000;
}

nav {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.titleNav_zeldaIconMobile {
  width: 2rem;
}

.titleNav__zeldaIconDesktop {
  display: none;
}

h1 {
  font-size: 1.2rem;
}

nav button img {
  width: 0.8rem;
  margin-left: 1.2em;
  transition: all 0.3s ease-in-out;
}

nav button img:hover {
  rotate: 90deg;
  transform: scale(1.5);
}

nav button {
  background: none;
  width: 3rem;
  border: none;
  cursor: pointer;
  padding-left: 0.8rem;
}

.titleNav p {
  font-size: 0.5rem;
  bottom: -10px;
  position: relative;
  left: 5px;
}

header nav a {
  display: flex;
}

@media screen and (min-width: 768px) {
  header {
    background-color: transparent;
    padding: 0.4rem;
    position: relative; 
    z-index: 1;
  }
  
  nav {
    justify-content: center;
  }
  
  .titleNav_zeldaIconMobile {
    display: none;
  }

  .titleNav__zeldaIconDesktop {
    display: inline;
    width: 60%;
  }

  .titleNav p {
    font-size: 1.5rem;
    bottom: -280px;
    color: #a08852;
    position: relative;
    left: 5px;
  }
  
  h1 {  
    display: none;  
  } 
  
  nav button img {
    display:none;
  }
  
  nav button img:active {
  margin-left: 1rem;
  }
}
