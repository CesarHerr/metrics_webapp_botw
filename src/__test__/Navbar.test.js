import '@testing-library/jest-dom';
import { screen, render } from '@testing-library/react';
import Navbar from '../components/Navbar';
import { MemoryRouter } from 'react-router-dom';

describe('Categories', () => {
  it('Renders Categories component', () => {
    const nav = render(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>
    );

    expect(nav.container).toMatchSnapshot();
  });

  it('Renders Navigator component', () => {
    render(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>
    );

    const element = screen.getByText('Zelda - Botw');

    expect(element).toBeInTheDocument();
  });
});