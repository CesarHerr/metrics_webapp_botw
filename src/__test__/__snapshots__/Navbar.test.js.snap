// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories Renders Categories component 1`] = `
<div>
  <header>
    <nav>
      <button
        class="nav-button"
        type="button"
      >
        <img
          alt="master sword"
          src="masterSword4.png"
        />
      </button>
      <a
        aria-current="page"
        class="titleNav active"
        href="/"
      >
        <h1>
          Zelda - Botw
        </h1>
        <p>
           ðŸ“šEncyclopedia
        </p>
      </a>
      <img
        alt="zelda icon"
        class="zeldaIcon"
        src="zeldaicon.png"
      />
    </nav>
  </header>
</div>
`;
